custom_loop_analysis.cpp

```diff
@@ -175,6 +179,30 @@ static llvm::Statistic NumLoopsNoStore = {"", "NumLoopsNoStore", "subset of loop
 static llvm::Statistic NumLoopsNoLoad = {"", "NumLoopsNoLoad", "subset of loops that has no Load instructions"};
 static llvm::Statistic NumLoopsWithCall = {"", "NumLoopsWithCall", "subset of loops that has a call instructions"};
 
+static void AddMetadataToBackEdge(LLVMContext &Ctx, BasicBlock *BB){
+    for (BasicBlock::iterator I = BB->begin(); I != BB->end(); ++I){
+        //counter++;
+        Instruction &i = *I;
+        if (isa<BranchInst>(i)){
```

Comment:
There could be multiple branch instructions. Verify that the branch target is in fact the loop header (body?)